src/app.rs

```diff
@@ -301,135 +278,67 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
         }
     }
 
-    pub fn patch_window_event_handlers(&self) {
-        if let Some(window_events) = self.cfg.window_events {
-            let new_event_handlers = (window_events)(self.data.model.borrow().as_ref().unwrap());
-            let new_manager = EventHandlerManager::with_event_handlers(new_event_handlers);
-
-            let mut old_manager = self.data.window_event_handler_manager.replace(new_manager);
-            let mut new_manager = self.data.window_event_handler_manager.borrow_mut();
-
-            new_manager.attach_listeners(util::window(), Some(&mut old_manager), &self.mailbox());
-        }
-    }
-
-    pub fn add_message_listener<F>(&self, listener: F)
-    where
-        F: Fn(&Ms) + 'static,
-    {
-        self.data
-            .msg_listeners
-            .borrow_mut()
-            .push(Box::new(listener));
-    }
-
-    #[allow(clippy::too_many_arguments)]
-    pub(super) fn new(
-        update: UpdateFn<Ms, Mdl, INodes, GMs>,
-        sink: Option<SinkFn<Ms, Mdl, INodes, GMs>>,
-        view: ViewFn<Mdl, INodes>,
-        mount_point: Element,
-        routes: Option<RoutesFn<Ms>>,
-        window_events: Option<WindowEventsFn<Ms, Mdl>>,
-        init_cfg: OptDynInitCfg<Ms, Mdl, INodes, GMs>,
-        base_path: Rc<Vec<String>>,
-    ) -> Self {
-        let window = util::window();
-        let document = window.document().expect("get window's document");
-
-        Self {
-            init_cfg,
-            cfg: Rc::new(AppCfg {
-                document,
-                mount_point,
-                update,
-                sink,
-                view,
-                window_events,
-                base_path,
-            }),
-            data: Rc::new(AppData {
-                model: RefCell::new(None),
-                // This is filled for the first time in run()
-                main_el_vdom: RefCell::new(None),
-                popstate_closure: RefCell::new(None),
-                hashchange_closure: RefCell::new(None),
-                routes: RefCell::new(routes),
-                window_event_handler_manager: RefCell::new(EventHandlerManager::new()),
-                sub_manager: RefCell::new(SubManager::new()),
-                msg_listeners: RefCell::new(Vec::new()),
-                scheduled_render_handle: RefCell::new(None),
-                after_next_render_callbacks: RefCell::new(Vec::new()),
-                render_info: Cell::new(None),
-            }),
-        }
-    }
-
     /// Bootstrap the dom with the vdom by taking over all children of the mount point and
     /// replacing them with the vdom if requested. Will otherwise ignore the original children of
     /// the mount point.
```

Comment:
Update this documentation, since it seems like the optional part has changed. Also, add notes about how it's to replace the vdom. Might also be good to note that this is only run once, at startup (or it should be? If it doesn't that something else probably is wrong).